File: backend/src/main/java/by/bsu/cardirectory/entity/Manufacturer.java
package by.bsu.cardirectory.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.OnDelete;

import java.time.LocalDate;
import java.util.List;

@Entity
@Data
public class Manufacturer {

    @Id
    @SequenceGenerator(
            name = "manufacturer_id_seq",
            sequenceName = "manufacturer_id_seq",
            allocationSize = 1
    )
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "manufacturer_id_seq"
    )
    private Long id;

    @Column(unique = true)
    private String name;

    private String country;

    private LocalDate foundationDate;

    private Integer employeesNumber;

    @OneToMany(
            cascade = { CascadeType.REFRESH, CascadeType.MERGE },
            targetEntity = Car.class,
            mappedBy = "manufacturer",
            fetch = FetchType.EAGER
    )
    private List<Car> cars;
}

File: frontend/src/interfaces/manufacturer.ts
import { INamedEntity } from './entity';
import { ICar } from './car';

export interface IManufacturer extends INamedEntity {
  name: string;
  country: string;
  foundationDate: string;
  employeesNumber: number;
  cars?: ICar[];
}

export type IManufacturerValues = IManufacturer[keyof IManufacturer];

export interface IManufacturerPayload {
  name: string;
  country: string;
  foundationDate: string;
  employeesNumber: number;
  cars: string[];
}
